libcxx/include/__format/formatter_integral.h

```diff
@@ -49,7 +51,8 @@ namespace __formatter {
 // Generic
 //
 
-_LIBCPP_HIDE_FROM_ABI inline char* __insert_sign(char* __buf, bool __negative, __format_spec::__sign __sign) {
+template <contiguous_iterator _Iterator>
```

Comment:
These iterators should be constrained to have a `char` `value_type` the current constrains loosen these restrictions. (I'm fine with new concept `contiguous_char_iterator` if you prefer that.)